src/index/sparse/sparse_inverted_index.h

```diff
@@ -118,6 +128,43 @@ class BaseInvertedIndex {
     n_cols() const = 0;
 };
 
+// CPU capability detection for SIMD optimizations
+#if defined(__x86_64__) || defined(_M_X64)
+namespace simd {
```

Comment:
consider move these simd codes to src/simd folder

Reply:
thank you moved all smid related code over